src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1217,6 +1207,23 @@ ActivateNodeList(List *nodeList)
 		}
 	}
 
+	/*
+	 * Ideally, we'd want to drop and sync the metadata inside a
+	 * single transaction. However, for some users, the metadata might
+	 * be excessively large. In that cases, Postgres fails to handle
+	 * processing of all the commands inside a single transaction.
+	 *
+	 * As of PG 15, we get error messages like the following at
+	 * commit time that are caused by relcache/catcache invalidations:
+	 *    ERROR:  invalid memory alloc request size 1073741824
+	 *
+	 * That's why, we try to split as much of the work as possible.
+	 * It incurs some risk of failures during these steps causing
+	 * issues if any of the steps is not idempotent.
+	 */
+	DropExistingMetadataInOutsideTransaction(nodeToSyncMetadata);
```

Comment:
I'm slightly wondering whether we should make it configurable (or at least easily adjustable in code)

Reply:
Let me see if I can do that easily